unit_tests/src/test_delay_hints_parser.cpp

```diff
@@ -262,36 +256,35 @@ TEST_CASE("#5503: getNextRepeatTime parses frequency correctly")
            {
                 .expected_action_code = 4,
                 .expected_next_time   = "10",
-                .expected_frequency   = double_until_success_or_until_time_string.replace(0, 2, "20")
+                .expected_frequency   = double_until_success_or_until_time_string_result
            }
         }},
         {"DOUBLE UNTIL SUCCESS OR <N> TIMES", {
             // Input
            {
                 .current_time = "0",
-                .frequency    = double_until_success_or_n_times
+                .frequency    = "10s DOUBLE UNTIL SUCCESS OR 20 TIMES"
            },
             // Output
            {
-                .expected_action_code = 0,
-                .expected_next_time   = "20",
-                .expected_frequency   = double_until_success_or_n_times
+                .expected_action_code = 4,
+                .expected_next_time   = "10",
+                .expected_frequency   = "20s DOUBLE UNTIL SUCCESS OR 19 TIMES. ORIGINAL TIMES=20"
            }
         }},
-        {"DOUBLE UNTIL SUCCESS OR UPTO <TIME>", {
+        {"DOUBLE UNTIL SUCCESS UPTO <TIME>", {
```

Comment:
What lead to this change?

If `OR` is incorrect, then we will need to update the docs as well. See https://docs.irods.org/4.3.0/plugins/pluggable_rule_engine/#delay-execution

Reply:
The current implementation and the test were in disagreement. What ended up happening was I 'sided' with the existing implementation, probably cause I thought that it was an error in the test since I was already working on the previous ones.

I think I also looked at the 4.3.0 doc you provided too, and didn't see any reference to an `UPTO` directive. For consistency though, it probably should have an `OR`.